### PR TITLE
concerning #818: applied requested change in Repository.IsValid specification to:

### PR DESCRIPTION
**Pull request by ptr1120**
* throw on null arguments
* return false on string.Empty

### CODE REVIEW

File: LibGit2Sharp/Repository.cs [hunk 1]
```diff
@@ -155,7 +155,12 @@ public Repository(string path, RepositoryOptions options)
         /// <returns>True if a repository can be resolved through this path; false otherwise</returns>
         static public bool IsValid(string path)
         {
-            Ensure.ArgumentNotNullOrEmptyString(path, "path");
+            // should only throw on null arguments
+            Ensure.ArgumentNotNull(path, "path");
+            if (path.Trim() == string.Empty)
```
Comment (nulltoken) [inline LibGit2Sharp/Repository.cs]: How about `string.IsEmptyOrWhiteSpaces()`?

**Comment by nulltoken** [comment]
Fix #818

File: LibGit2Sharp/Repository.cs (hunk 1 above)
Comment (ptr1120) [inline LibGit2Sharp/Repository.cs]: ohh, I forgot, thanks....

**Comment by nulltoken** [comment]
:cool:

One final nitpick. Could you please squash those two commits together (we don't merge review fixup commits) and maybe simplify your commit message?

Maybe something like

```
Make Repository.IsValid() return false on empty paths

Fix #818
```

The third line should allow to dynamically close the issue when this PR is merged.

**Comment by nulltoken** [comment]
FWIW, the previous request will rewrite your branch history. Once done, just force push your branch, this will automatically update this PR.

**Pull request closed without merging**